torchgeo/trainers/classification.py

```diff
@@ -37,7 +30,9 @@ def __init__(
         model: str = 'resnet50',
         weights: WeightsEnum | str | bool | None = None,
         in_channels: int = 3,
-        num_classes: int = 1000,
+        task: str = 'multiclass',
+        num_classes: int | None = None,
+        num_labels: int | None = None,
```

Comment:
The defaults here match torchmetrics. If `task='multiclass'`, only `num_classes` is used. If `task='multilabel'`, only `num_labels` is used. If `task='binary'`, both are ignored. Honestly, we could have a single `num_classes` if we want and simply use it for both.